earth_enterprise/src/fusion/portableglobe/gerewritedbroot.cpp

```diff
@@ -219,21 +225,28 @@ void ReplaceReferencedKml(const std::string &new_kml_base_url,
       if (string_id_or_value->has_value()) {
         // the url is stored directly in the kml_url field
         std::string kml_url = string_id_or_value->value();
-        fout << kml_url << " " << kml_file_name << std::endl;
+        if ( preserve_kml_filenames ) {
```

Comment:
I agree that that's an edge case that we should account for - thanks for catching this.
Google Documentation on this recommends only have a single KML in the database and adding links from that KML to other network linked KMLs. (http://www.opengee.org/geedocs/5.2.0/answer/2760457.html). Since this use case falls outside of the recommended practice, I recommend logging this as a separate issue.

Reply:
In offline discussion, @tst-ppenev and I agree that we'll write up another ticket to address the edge case of filename collisions.